src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -81,29 +86,33 @@ default FilePath getFilePathForUniqueFolder(Launcher launcher, String uniqueName
     default String getUniqueNameForRunnerFile() {
         //Using 8 bit long random alphanumeric string
         return RandomStringUtils.randomAlphanumeric(8);
-    }
+    } 
     
     // This method prepares the temp folder by coping all helper files in it.
-    default void prepareTmpFldr(FilePath tmpFldr, String runnerScript) throws IOException, InterruptedException {
+       default void prepareTmpFldr(FilePath tmpFldr, String runnerScript) throws IOException, InterruptedException {
+    	// copy genscript package
+    	copyFileInWorkspace(MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR,
+                 MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR, tmpFldr);
+    	FilePath zipFileLocation =
+                 new FilePath(tmpFldr, MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR);
+    	runnerScript=getZipScript(runnerScript, zipFileLocation.getRemote());
         // Write MATLAB scratch file in temp folder.
         FilePath scriptFile =
-                new FilePath(tmpFldr, getValidMatlabFileName(tmpFldr.getBaseName()) + ".m");
+                 new FilePath(tmpFldr, getValidMatlabFileName(tmpFldr.getBaseName()) + ".m");
         scriptFile.write(runnerScript, "UTF-8");
-        // copy genscript package
-        copyFileInWorkspace(MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR,
-                MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR, tmpFldr);
-        FilePath zipFileLocation =
-                new FilePath(tmpFldr, MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR);
-
-        // Unzip the file in temp folder.
-        zipFileLocation.unzip(tmpFldr);
-    }
+ }
     
+    default String getZipScript(String script, String url) {

```

Comment:
`getZipScript` seems like a strange name for this function. It seems like this function is just replacing a placeholder in the a string. Perhaps something like `replaceZipPlaceholder`?